tests/integration/it_employees_test.py

```diff
@@ -146,12 +150,30 @@ def test_cannot_access_create(client, db_session):
     assert client.get("/employees/create").status_code == HTTPStatus.FORBIDDEN
 
 
-@pytest.mark.skip(reason="Authentication injection not implemented")
+"""
+@todo #412:30min timeless/employees/views.py::Delete is not working. Fix it and 
+ then uncomment the tests below. Also refactor Delete to EmployeeDeleteView to
+ make it uniform with the rest of the applicaton. 
+"""
+
+
+@pytest.mark.skip(reason="Correct EmployeeDeleteView")
 def test_delete(client):
-    employee = factories.EmployeeFactory()
-    response = client.post(url_for('employee.delete', id=employee.id))
-    assert response.status_code == HTTPStatus.FOUND
-    assert not Employee.query.count()
+    company = factories.CompanyFactory()
+    intern = factories.EmployeeFactory(
+        company=company, role_id=Role(role_type=RoleType.Intern.name).id
+    )
+    boss = factories.EmployeeFactory(
+        company=company, role_id=Role(role_type=RoleType.Manager.name).id
+    )
+    with client.session_transaction() as session:
+        session["user_id"] = boss.id
+    g.user = boss
+    intern_id = intern.id
+    response = client.post(url_for("employee.delete", id=intern_id))
+    assert len(Employee.query.all()) == 1
```

Comment:
@paulodamaso 
`assert Employee.query.count() == 1` will not build model instances as opposed to taking `len`